From 75db995dda74696123cde2869e4e9b3315b91abd Mon Sep 17 00:00:00 2001
From: "Yann E. MORIN" <yann.morin.1998@free.fr>
Date: Sun, 1 Aug 2021 13:40:28 +0200
Subject: [PATCH] configure: accept non-system cups-config

In cross-compilation setups, cups.config might not be installed, and
might not be in the PATH.

Allow a user to specify the full path to cups-config.

Signed-off-by: Yann E. MORIN <yann.morin.1998@free.fr>
---
 configure.ac | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index dd013b4a..9f033d18 100644
--- a/configure.ac
+++ b/configure.ac
@@ -33,7 +33,11 @@ fi
 AC_SUBST(DESKTOPVENDOR)
 AC_SUBST(DESKTOPPREFIX)
 
-cupsserverbindir="`cups-config --serverbin`"
+AC_ARG_WITH([cups-config],
+            [AC_HELP_STRING([--with-cups-config], [Specify the path of cups-config])],,
+            [with_cups_config=cups-config])
+
+cupsserverbindir="`${with_cups_config} --serverbin`"
 AC_SUBST(cupsserverbindir)
 
 PKG_CHECK_MODULES(GLIB, glib-2.0, has_glib=yes, has_glib=no)
-- 
2.25.1

